### PR TITLE
Add second example and documentation

### PR DESCRIPTION
**Pull request by HabeebaMinhar**
## Purpose

Fixes:$subject

## Checklist

- [ ] Linked to an issue
- [ ] Updated the specification
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Checked native-image compatibility

### CODE REVIEW

File: examples/paper-invoice-to-inspectionlot/main.bal [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+const OCR_REQUEST_PATH = "/invoice_parser";
+const OCR_URL = "https://api.edenai.run/v2/ocr";
+const CONTENT_TYPE = "Content-Type";
+const APPLICATION_JSON = "application/json";
+
+configurable S4HANAClientConfig s4hanaClientConfig = ?;
+configurable string ocrToken = ?;
+configurable string invoiceUrl = ?;
+
+final map<string> & readonly supplierMap = {"SNG Engineering Inc": "1010"};
```
Comment (LakshanWeerasinghe) [inline examples/paper-invoice-to-inspectionlot/main.bal]: can make these maps constants.

File: examples/paper-invoice-to-inspectionlot/main.bal [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+);
+
+public function main() {
+
+    PaperInvoice|error invoiceResponse = readPaperInvoice();
+    if invoiceResponse is error {
+        log:printError("Error while reading paper invoice: " + invoiceResponse.message());
+        return;
+    }
+    //io:println(invoiceResponse);
```
Comment (LakshanWeerasinghe) [inline examples/paper-invoice-to-inspectionlot/main.bal]:
```suggestion
```

File: examples/paper-invoice-to-inspectionlot/types.bal [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    string Material;
+    string Plant;
+    int Quantity;
+};
+
+type S4HANAClientConfig record {|
+    string hostname;
+    string username;
+    string password;
+|};
```
Comment (LakshanWeerasinghe) [inline examples/paper-invoice-to-inspectionlot/types.bal]:
```suggestion
|};

```

File: examples/paper-invoice-to-inspectionlot/main.bal [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+isolated function readPaperInvoice() returns PaperInvoice|error {
+    ExtractedInvoice|http:Error response = ocrHttpClient->post(
+        path = OCR_REQUEST_PATH,
+        message = {
+        show_original_response: "false",
+        fallback_providers: "",
+        providers: "google",
+        language: "en",
+        file_url: invoiceUrl
+    },
```
Comment (LakshanWeerasinghe) [inline examples/paper-invoice-to-inspectionlot/main.bal]: format this file again.

File: examples/paper-invoice-to-inspectionlot/Ballerina.toml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+[package]
+org = "habeebaminhar"
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/Ballerina.toml]:
```suggestion
org = "wso2"
```

File: examples/paper-invoice-to-inspectionlot/Extract data from paper invoice to create Inspection lot.md.md [hunk 6]
```diff
@@ -0,0 +1,41 @@
+# Extract Paper Invoice Data and Create Inspection Lot
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/Extract data from paper invoice to create Inspection lot.md.md]:
```suggestion
# Extract paper Delivery Note and integrate with S/4HANA
```

File: examples/paper-invoice-to-inspectionlot/Extract data from paper invoice to create Inspection lot.md.md [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+
+In the example directory, create a Config.toml file and configure InspectionLot and S/4HANA API credentials as shown below:
+
+```toml
+[s4hanaClientConfig]
+hostname = "<Hostname>"
+username = "<Username>"
+password = "<Password>"
+
+[<ORG_NAME>.create-inspectionlot-from-paper-invoice]
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/Extract data from paper invoice to create Inspection lot.md.md]: Is this necessary

File: examples/paper-invoice-to-inspectionlot/main.bal [hunk 8]
```diff
@@ -0,0 +1,106 @@
+import ballerina/http;
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/main.bal]: License

File: examples/paper-invoice-to-inspectionlot/main.bal [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+
+public function main() {
+
+    PaperInvoice|error invoiceResponse = readPaperInvoice();
+    if invoiceResponse is error {
+        log:printError("Error while reading paper invoice: " + invoiceResponse.message());
+        return;
+    }
+
+    log:printInfo("Received inspection lot with invoice number: " + invoiceResponse.invoice_number + " on " + invoiceResponse.date);
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/main.bal]:
```suggestion
    log:printInfo("Received inspection lot with invoice number: " + invoiceResponse.invoice_number + " on " + invoiceResponse.date);
    
```

File: examples/paper-invoice-to-inspectionlot/main.bal [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+    io:print(inspectionLotData);
+
+    InspectionLot[] data = inspectionLotData._InspectionLotDetails;
+    inspectionLot:CreateA_InspectionLot inspectionLot = {
+        InspectionLot: inspectionLotData.InspectionLot,
+        InspectionLotType: inspectionLotData.InspectionLotType,
+        Material: data[0].Material,
+        Plant: data[0].Plant,
+        InspectionLotQuantity: data[0].Quantity.toString()
+    };
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/main.bal]: Isnt this the transform code? Why are we transforming twice?

File: examples/paper-invoice-to-inspectionlot/main.bal [hunk 11]
```diff
@@ -0,0 +1,106 @@
...
+
+    inspectionLot:A_InspectionLotWrapper|error inspectionLotResponse = inspectionLotClient->createA_InspectionLot(inspectionLot);
+    if inspectionLotResponse is error {
+        log:printError("Error while creating SAP Inspection Lot: " + inspectionLotResponse.message());
+    }
+     io:println(inspectionLotResponse);
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/main.bal]: Print a useful log saying the inpection lot with is is printed

File: examples/paper-invoice-to-inspectionlot/main.bal [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+isolated function getOCRHttpClient() returns http:Client|error {
+    http:BearerTokenConfig tokenAuthHandler = {token: ocrToken};
+    return new (url = OCR_URL, auth = tokenAuthHandler, cookieConfig = {enabled: true});
+}
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/main.bal]: Can be done globally

File: examples/paper-invoice-to-inspectionlot/types.bal [hunk 13]
```diff
@@ -0,0 +1,35 @@
+type ExtractedInvoice record {
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/types.bal]: License

File: examples/paper-invoice-to-inspectionlot/types.bal [hunk 14]
```diff
@@ -0,0 +1,35 @@
...
+};
+
+type PaperInvoice record {
+    string invoice_number;
+    string date;
+    record {string merchant_name;} merchant_information;
+    InvoiceItem[] item_lines;
+};
+
+type InvoiceItem record {
```
Comment (niveathika) [inline examples/paper-invoice-to-inspectionlot/types.bal]: Cant we reuse the types within the package

**Review by niveathika — CHANGES_REQUESTED** [review]

File: examples/paper-invoice-to-inspectionlot/main.bal (hunk 10 above)
Comment (HabeebaMinhar) [reply examples/paper-invoice-to-inspectionlot/main.bal]: Made the [improvement](https://github.com/ballerina-platform/module-ballerinax-sap.s4hana.qm/pull/24/commits/230ab97a4f3ea8e5edb4540319d92a08570c17ee)

File: examples/paper-invoice-to-inspectionlot/Extract data from paper invoice to create Inspection lot.md.md (hunk 7 above)
Comment (HabeebaMinhar) [reply examples/paper-invoice-to-inspectionlot/Extract data from paper invoice to create Inspection lot.md.md]: Updated this and it is necessary

**Review by niveathika — APPROVED** [review]

**Pull request merged**